File: ws_slam/src/my_py_pkg/my_py_pkg/number_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from std_msgs.msg import Int32

class NumberPublisher(Node):
    def __init__(self):
        super().__init__('number_publisher')
        self.publisher_ = self.create_publisher(Int32, 'number', 10)
        self.timer = self.create_timer(1.0, self.publish_number)
        self.number = 3
        self.get_logger().info(f'Number Publisher started, publishing {self.number}')

    def publish_number(self):
        msg = Int32()
        msg.data = self.number
        self.publisher_.publish(msg)
        self.get_logger().info(f'Published: {msg.data}')

def main(args=None):
    rclpy.init(args=args)
    node = NumberPublisher()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: ws_slam/src/my_cpp_pkg/src/number_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/int32.hpp"

using namespace std::chrono_literals;

class NumberPublisher : public rclcpp::Node
{
public:
    NumberPublisher()
    : Node("number_publisher"), number_(3)
    {
        publisher_ = this->create_publisher<std_msgs::msg::Int32>("number", 10);
        timer_ = this->create_wall_timer(1s, std::bind(&NumberPublisher::publish_number, this));
        RCLCPP_INFO(this->get_logger(), "Number Publisher started, publishing: %d", number_);
    }

private:
    void publish_number()
    {
        auto msg = std_msgs::msg::Int32();
        msg.data = number_;
        publisher_->publish(msg);
        RCLCPP_INFO(this->get_logger(), "Published: %d", msg.data);
    }

    rclcpp::Publisher<std_msgs::msg::Int32>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
    int number_;
};

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<NumberPublisher>());
    rclcpp::shutdown();
    return 0;
}
